pkg/controller/migplan/migplan_controller.go

```diff
@@ -532,87 +529,100 @@ func (r ReconcileMigPlan) waitForMigAnalyticsReady(plan *migapi.MigPlan) (*migap
 	migAnalytics := &migapi.MigAnalyticList{}
 	err := r.List(context.TODO(), k8sclient.MatchingLabels(map[string]string{MigPlan: plan.Name}), migAnalytics)
 	if err != nil {
-		return nil, err
+		return nil, liberr.Wrap(err)
 	}
-	for _, migAnalytic := range migAnalytics.Items {
+	for i := range migAnalytics.Items {
+		migAnalytic := &migAnalytics.Items[i]
 		if migAnalytic.Spec.AnalyzeExtendedPVCapacity == true {
-			if migAnalytic.Status.IsReady() {
-				return &migAnalytic, nil
+			if migAnalytic.Status.IsReady() && !migAnalytic.Spec.Refresh {
+				return migAnalytic, nil
 			}
 			if time.Now().Sub(migAnalytic.CreationTimestamp.Time) > migAnalyticsTimeout {
 				plan.Status.SetCondition(migapi.Condition{
-					Type:     IntelligentPVResizingDisabled,
+					Type:     PVResizingDisabled,
```

Comment:
I think `PVUsageAnalysisFailed` or `PVUsageDetectionFailed` or `PVSizeDetectionFailed` would be a more appropriate name for this condition. When I first saw it, I thought this was related to whether the feature flag was set.

Reply:
+1